internal/command/test.go

```diff
@@ -1065,8 +1066,18 @@ func getEvalContextFromStates(states map[string]*TestFileState) *hcl.EvalContext
 		}
 		runCtx[state.Run.Name] = cty.ObjectVal(outputs)
 	}
-	ctx := &hcl.EvalContext{Variables: map[string]cty.Value{"run": cty.ObjectVal(runCtx)}}
 
+	varCtx := make(map[string]cty.Value)
+	for _, variable := range config.Module.Variables {
+		varCtx[variable.Name] = variable.Default
+	}
```

Comment:
This way we **only** use the default value of a variable. If a variable was explicitly set in the `tofu test` run, via a tfvars file of via a `TF_VAR_var1` environment variable, this value is completely ignored, and only the default value is used.

We should change this behaviour to allow for `TF_VAR_var1=false tofu test` to be passed down to the `variables` block in `var.` references

Reply:
I have revised the code and shared my approach [here.](https://github.com/opentofu/opentofu/pull/1488#issuecomment-2117165831)